src/transformers/models/big_bird/modeling_big_bird.py

```diff
@@ -436,6 +437,69 @@ def __init__(self, config, seed=None):
         self.key = nn.Linear(config.hidden_size, self.all_head_size, bias=config.use_bias)
         self.value = nn.Linear(config.hidden_size, self.all_head_size, bias=config.use_bias)
 
+        self.rand_attn_tables = [[] for _ in range(config.max_position_embeddings // config.block_size - 1)]
+        self.generate_rand_attn_tables(
+            self.max_seqlen, self.max_seqlen, self.block_size, self.block_size, self.num_random_blocks, 1024
+        )
+        self.rand_attn_tables_prepared_arg = [
+            self.max_seqlen,
+            self.max_seqlen,
+            self.block_size,
+            self.block_size,
+            self.num_random_blocks,
+            1024,
+        ]
+
+    def generate_one_table(self, o_table, start_i, end_i, num_rand_blocks):
+        return list(permutations(o_table[start_i:end_i], num_rand_blocks))
```

Comment:
Not a real fan of a oneliner like this. Also if we indeed kep the function, let's give some hints on what the shapes are. 
Here you have a list of length ( factorial(end_i - start_i)), where each element is of length (end_i - start_i).

Reply:
add comments in https://github.com/huggingface/transformers/pull/21499/commits/4335620f0fbfd1aa50b0955fa2a62d419bb3524b